Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -2,17 +2,157 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include "Core/GeckoCodeConfig.h"
+
 #include <algorithm>
 #include <sstream>
 #include <string>
 #include <vector>
 
+#include <curl/curl.h>
+
 #include "Common/IniFile.h"
+#include "Common/Logging/Log.h"
 #include "Common/StringUtil.h"
-#include "Core/GeckoCodeConfig.h"
 
 namespace Gecko
 {
+static size_t DownloadCodesWriteCallback(void* contents, size_t size, size_t nmemb,
+                                         std::string* body)
+{
+  size_t realsize = size * nmemb;
+  body->insert(body->end(), reinterpret_cast<char*>(contents),
+               reinterpret_cast<char*>(contents) + realsize);
+  return realsize;
+}
+
+std::vector<GeckoCode> DownloadCodes(std::string gameid, bool* succeeded)
+{
+  switch (gameid[0])
+  {
+  case 'R':
+  case 'S':
+  case 'G':
+    break;
+  default:
+    // All channels (WiiWare, VirtualConsole, etc) are identified by their first four characters
+    gameid = gameid.substr(0, 4);
+    break;
+  }
+
+  CURL* curl = curl_easy_init();
```

Comment:
This should likely be:

```cpp
std::unique_ptr<CURL, decltype(&curl_easy_cleanup)> curl{curl_easy_init(), curl_easy_cleanup};
```

So it automatically does cleanup/freeing when it goes out of scope.